python/dgl/graphbolt/graph_storage/csc_sampling_graph.py

```diff
@@ -273,8 +279,27 @@ def sample_neighbors(
             assert len(self.metadata.edge_type_to_id) == fanouts.size(
                 0
             ), "Fanouts should have the same number of elements as etypes."
+        if probs is not None:
+            assert probs.dim() == 1, "Probs should be 1-D tensor."
+            assert (
+                probs.size(0) == self.num_edges
+            ), "Probs should have the same number of elements as the number \
+                of edges."
+            assert probs.dtype in [
+                torch.bool,
+                torch.float16,
+                torch.bfloat16,
+                torch.float32,
+                torch.float64,
+            ], "Probs should have a floating-point or boolean data type."
+            # Note probs will be passed as input for 'torch.multinomial'
+            # in deeper stack, which doesn't support 'torch.half' and
+            # torch.bool' data types.To avoid crashes, convert 'probs' to
+            # 'bfloat16' data type.
+            if probs.dtype in [torch.bool, torch.float16]:
+                probs = probs.to(torch.bfloat16)
```

Comment:
Put this to c++ because it is tied to your underlying implementation instead of the python operator semantics.

Reply:
Fixed